### PR TITLE
Add support for the Graph API that is part of xpack

### PR DESCRIPTION
**Pull request by Mpdreamz**
https://www.elastic.co/guide/en/graph/2.3/graph-introduction.html

### CODE REVIEW

**Comment by gmarz** [comment]
LGTM :+1:

File: src/Nest/XPack/Graph/Explore/Request/GraphExploreControls.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		[JsonProperty("timeout")]
+		Time Timeout { get; set; }
+
+		[JsonProperty("sample_diversity")]
+		SampleDiversity SampleDiversity { get; set; }
+	}
+
+	public class GraphExploreControls : IGraphExploreControls
+	{
+		[JsonProperty("use_significance")]
```
Comment (russcam) [inline src/Nest/XPack/Graph/Explore/Request/GraphExploreControls.cs]: Do we need the `JsonProperty` attributes on the concrete type and the interface?

**Comment by russcam** [comment]
Just a couple of small things:

- Included the GraphExploreControls in the integration test
- Removed the `JsonProperty` attributes from the concrete `GraphExploreControls`

LGTM :+1:

**Comment by Mpdreamz** [comment]
Manually cherry-picked to 2.x and master

**Pull request closed without merging**